e2e: node: fix serial lanes

ffromani opened the pull request:
#### What type of PR is this?

/kind cleanup
/kind failing-test

#### What this PR does / why we need it:

- Fix the node container manager tests to safely reset the cpumanager policy using the existing, exercised utilities vs roll another (apparently buggy) variant.
- Fix the podresources tests to avoid panic for nil map when setting a feature gate


#### Which issue(s) this PR is related to:
133314

#### Special notes for your reviewer:
Collecting all the fixes in this PR

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

ffromani committed:
e2e: node: cleanup state file on cpumanager policy reset

When we transition from static to none cpumanager policy,
we always need to drain the node, or, in test, to remove
the state files. This is a very longstanding cpumanager constraint.

In practice, we have helpers to do this safely, tested and exercised
in the cpumanager e2e tests;
let's make use of them in the rest of the e2e_node tests.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani commented:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-swap-conformance-fedora-serial
/test pull-kubernetes-node-swap-ubuntu-serial

ffromani commented:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-swap-conformance-fedora-serial
/test pull-kubernetes-node-swap-ubuntu-serial

kannon92 commented:
> /test pull-kubernetes-node-swap-ubuntu-serial

I0731 12:39:49.279974   11762 feature_gate.go:385] feature gates: {map[KubeletServiceAccountTokenForCredentialProviders:true NodeSwap:true]}
E0731 12:39:49.280154   11762 run.go:72] "command failed" err="failed to validate kubelet configuration, error: invalid configuration: can't use reservedSystemCPUs (--reserved-cpus) with systemReservedCgroup (--system-reserved-cgroup) or kubeReservedCgroup (--kube-reserved-cgroup), path: &TypeMeta{Kind:,APIVersion:,}"

Looks like this reproduces the other issue.

kannon92 commented:
/test

ffromani commented:
this will take longer than expected. I'm proposing a 2-phase approach here: https://github.com/kubernetes/kubernetes/pull/133353

pacoxu commented:
/cc @kannon92 @SergeyKanzhelev
Is this a release blocker?
cc @kubernetes/release-team-leads @Vyom-Yadav

ffromani committed:
e2e: node: podresources: avoid panic for nil map

always ensure the FeatureGate map exists before to set a FeatureGate

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani commented:
I still can't reproduce the failures we see in CI locally. I'm having little success and finding only part of the issues.

ffromani committed:
e2e: node: device: fix permafailing device plugin test

A device plugin test was incorrectly depending too much
on a podresources API ancient bug fixed in https://github.com/kubernetes/kubernetes/pull/132028
Let's update the test to deal with the new and correct
behavior to make the test green again.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
WIP: e2e: node: serial: test fix for node container manager

WIP

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani commented:
/retest-required

ffromani commented:
/retest-required
